common/step_create_cdrom.go

```diff
@@ -107,25 +106,69 @@ func (s *StepCreateCD) Cleanup(multistep.StateBag) {
 	}
 }
 
-func retrieveCommandForOS(label string, source string, dest string) *exec.Cmd {
-	switch runtime.GOOS {
-	case "windows":
-		cmd := exec.Command("oscdimg")
-		args := []string{"-j1", "-o", "-m", "-l" + label, source, dest}
-		cmd.Args = append(cmd.Args, args...)
-		return cmd
-	case "darwin":
-		cmd := exec.Command("hdiutil")
-		args := []string{"makehybrid", "-o", dest, "-hfs",
-			"-joliet", "-iso", "-default-volume-name", label, source}
-		cmd.Args = append(cmd.Args, args...)
-		return cmd
-	default:
-		cmd := exec.Command("mkisofs")
-		args := []string{"-joliet", "-volid", label, "-o", dest, source}
-		cmd.Args = append(cmd.Args, args...)
-		return cmd
+type createCommand struct {
+	Name    string
+	Command func(path string, label string, source string, dest string) *exec.Cmd
+}
+
+var createCommands []createCommand = []createCommand{
+	{
+		"xorriso", func(path string, label string, source string, dest string) *exec.Cmd {
+			return exec.Command(
+				path,
+				"-as", "genisoimage",
```

Comment:
Indeed, `xorriso` does not need `genisoimage` installed, its just a mode of operation of `xorriso`.

About the macOS, if there is a `xorriso` command maybe the user really wants to use it because it has to be explicitly installed. But I'm OK with moving `xorriso` to the end of the list; just let me known.

Reply:
I think it's fine as a default.